api4/file.go

```diff
@@ -325,20 +325,25 @@ func getPublicFile(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if data, err := c.App.ReadFile(info.Path); err != nil {
+	if fileReader, err := c.App.FileReader(info.Path); err != nil {
 		c.Err = err
 		c.Err.StatusCode = http.StatusNotFound
-	} else if err := writeFileResponse(info.Name, info.MimeType, data, true, w, r); err != nil {
+	} else if err := writeFileResponse(info.Name, info.MimeType, info.Size, fileReader, true, w, r); err != nil {
 		c.Err = err
 		return
 	}
 }
 
-func writeFileResponse(filename string, contentType string, bytes []byte, forceDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
+func writeFileResponse(filename string, contentType string, contentSize int64, fileReader io.Reader, forceDownload bool, w http.ResponseWriter, r *http.Request) *model.AppError {
 	w.Header().Set("Cache-Control", "max-age=2592000, private")
-	w.Header().Set("Content-Length", strconv.Itoa(len(bytes)))
 	w.Header().Set("X-Content-Type-Options", "nosniff")
 
+	if contentSize > 0 {
+		w.Header().Set("Content-Length", strconv.Itoa(int(contentSize)))
+	} else {
+		w.Header().Set("Transfer-Encoding", "chunked")
```

Comment:
Not 100% on how this all fits together, but is setting this header strictly necessary? As per `net/http`:
> If Body is present, Content-Length is <= 0 and TransferEncoding hasn't been set to "identity", Write adds "Transfer-Encoding: chunked" to the header. Body is closed after it is sent.